### DISCUSSION TITLE
Cloudflare Proxied DNS & Uploading

### DISCUSSION THREAD

**Discussion by Geektasticdad** [advplyr/audiobookshelf#328] [Q&A]
I'm running this through Cloudflare DNS using NPM and Let's Encrypt wildcard SSL cert. Proxied host in NPM is pretty straight forward, have enabled Websocket Support and Block Common Exploits. SSL I've enabled Force SSl and HSTS support. No advanced settings or custom locations.

Here's the issue: Audiobookshelf runs fine with Proxied DNS enabled in Cloudflare EXCEPT uploading. It times out and I get "Oops something went wrong". Haven't found anything useful in the logs, just seems to time out. If I disable proxied DNS, uploads work fine.

Just wondering if anyone has any insight.

**Comment by advplyr** [comment]
That "Oops" log is in the code here: https://github.com/advplyr/audiobookshelf/blob/master/client/pages/upload/index.vue#L302

Can you open up the dev tools console and see what the full error is?

**Comment by duplaja** [comment]
Depending on what you're uploading, I know that the free version of Cloudflare has a 100mb max file upload limit (HTTP POST request size). Do you run into the same issues if you upload a single smaller file to test? I'm not sure if uploads are chunked, or sent as-is.

**Reply by ItsNoted** [reply]
I thought it was 1gb for free users? 100mb is nothing... that's kind of a let down.

**Reply by MidnightSnowleopard** [reply]
Nope, 100mb
https://developers.cloudflare.com/cache/about/default-cache-behavior/

**Comment by veno501** [comment]
I'm using Cloudflare tunneling with that upload limit and am getting a Payload too Large response. This is the notification that pops up in the web client:

```
<html>
<head><title>413 Payload Too Large</title></head>
<body>
<center><h1>413 Payload Too Large</h1></center>
<hr><center>cloudflare</center>
</body>
</html>
```


I believe the app should have the ability to chunk uploads to make payloads fit below Cloudflare's or whatever size limit the admin specifies

**Reply by H2OKing89** [reply]
If I remember right CF has a 50MB limit.
